lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenesisFiles.hs

```diff
@@ -221,50 +304,19 @@ generateGenesis initialFunds genesisMods = do
                         , sgStaking = Ledger.emptyGenesisStaking
                         , -- We need this to submit MIR certs
                           -- (and probably for the BFT node pre-babbage):
-                          sgGenDelegs =
-                            fromRight (error "invalid sgGenDelegs") . Aeson.eitherDecode
-                                $ Aeson.encode
-                                    [aesonQQ|
-                    {"91612ee7b158dc64871a959060973d0f2b8fb6e85ae960f03b8640ac": {
-                        "delegate": "180b3fae61789f61cbdbc69e5f8e1beae9093aa2215e482dc8d89ec9",
-                        "vrf": "e9ef3b5d81d400eb046de696354ff8e84122f505e706e3c86a361cce919a686e"
-                    }}|]
+                          sgGenDelegs = mempty
                         }
                     genesisMods
 
-        let byronFileOf, shelleyFileOf, alonzoFileOf, conwayFileOf
-                :: DirOf x  -> AbsFile
-            byronFileOf x = absDirOf x </> relFile "byron-genesis.json"
-            shelleyFileOf x = absDirOf x </> relFile "shelley-genesis.json"
-            alonzoFileOf x = absDirOf x </> relFile "alonzo-genesis.json"
-            conwayFileOf x = absDirOf x </> relFile "conway-genesis.json"
-
-        let shelleyGenesis = shelleyFileOf cfgClusterDir
-        Aeson.encodeFile (toFilePath shelleyGenesis) shelleyGenesisData
-
-        let fileToAeson :: FilePath -> IO Aeson.Value
-            fileToAeson f = Aeson.eitherDecodeFileStrict f >>= either fail pure
-
-        let byronGenesis = byronFileOf cfgClusterDir
-        fileToAeson (toFilePath $ byronFileOf cfgClusterConfigs)
-            >>= withAddedKey
-                "startTime"
-                (round @_ @Int $ utcTimeToPOSIXSeconds systemStart)
-            >>= Aeson.encodeFile (toFilePath byronGenesis)
-
-        let alonzoGenesis = alonzoFileOf cfgClusterDir
-        fileToAeson (toFilePath $ alonzoFileOf cfgClusterConfigs)
-            >>= Aeson.encodeFile (toFilePath alonzoGenesis)
-
-        let conwayGenesis = conwayFileOf cfgClusterDir
-        copyFile
-            (toFilePath $ conwayFileOf cfgClusterConfigs)
-            (toFilePath conwayGenesis)
-
-        pure
-            GenesisFiles
-                { byronGenesis = FileOf byronGenesis
-                , shelleyGenesis = FileOf shelleyGenesis
-                , alonzoGenesis = FileOf alonzoGenesis
-                , conwayGenesis = FileOf conwayGenesis
+        produceGenesis cfgClusterConfigs cfgClusterDir
+            $ GenesisRecord
+                { byronGenesis =
+                    Const
+                        $ key "startTime"
+                            .~ toJSON
+                                (round @_ @Int $ utcTimeToPOSIXSeconds systemStart)
+                , shelleyGenesis = Const
+                    $ \_ -> toJSON shelleyGenesisData
```

Comment:
So we could remove the shelley template? Or are you envisioning users being able to supply their own templates and the cluster startup doing only the minimal modifications?

Reply:
That would be my idea yes